libdnf/module/modulemd/ModuleMetadata.cpp

```diff
@@ -193,4 +193,32 @@ void ModuleMetadata::reportFailures(const GPtrArray *failures)
     }
 }
 
+ModulemdObsoletes * ModuleMetadata::getNewestActiveObsolete(ModulePackage *p)
+{
+    ModulemdModule * myModule = modulemd_module_index_get_module(resultingModuleIndex, p->getName().c_str());
+    if (myModule == nullptr) {
+        return nullptr;
+    }
+
+    GError *error = NULL;
+    ModulemdModuleStream * myStream = modulemd_module_get_stream_by_NSVCA(myModule,
```

Comment:
ModulePackage contains ModulemdModuleStream, why you search for it? Just to need to be a friend. Or did I overlooked something?

Reply:
Unfortunately the `ModulemdModuleStream` inside this `ModulePackage` was constructed with metadata from only one repository and the obsoletes may be missing. Its mentioned in a comment before this function is called in `applyObsoletes`. We have to find the stream in `pImpl->moduleMetadata` instance from `ModulePackageContainer` because it contains all available module metadata merged together.

Reply:
Ok, thanks for an explanation.